devices/chipdb_nexus.cmake

```diff
@@ -0,0 +1,101 @@
+function(create_prjoxide_device_db)
+    # ~~~
+    # create_prjoxide_device_db(
+    #    device <common device>
+    #    output_target <output device target>
+    # )
+    # ~~~
+    #
+    # Generates a device database from Project Oxide
+    #
+    # If output_target is specified, the output_target_name variable
+    # is set to the generated output_device_file target.
+    #
+    # Arguments:
+    #   - device: common device name of a set of parts. E.g. LIFCL-17
+    #   - output_target: variable name that will hold the output device target for the parent scope
+    #
+    # Targets generated:
+    #   - prjoxide-<device>-device
+    set(options)
+    set(oneValueArgs device output_target)
+    set(multiValueArgs)
+
+    cmake_parse_arguments(
+        create_prjoxide_device_db
+        "${options}"
+        "${oneValueArgs}"
+        "${multiValueArgs}"
+        ${ARGN}
+    )
+
+    set(device ${create_prjoxide_device_db_device})
+    set(output_target ${create_prjoxide_device_db_output_target})
+    set(prjoxide_device_db ${CMAKE_CURRENT_BINARY_DIR}/${device}.device)
+    add_custom_command(
+        OUTPUT ${prjoxide_device_db}
+        COMMAND
+            ${PRJOXIDE}
```

Comment:
`PRJOXIDE`'s binary should be present in the `programs` target as a property. You may get like [this](https://github.com/SymbiFlow/fpga-interchange-tests/blob/d244771a4a618c9aa686f8e0f44698f1bcf4da8a/tests/tests.cmake#L72) (and probably avoid to pass a CMake parameter

Reply:
Done, thanks